osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/BundleMessageProcessorTest.java

```diff
@@ -0,0 +1,122 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.net.ConnectException;
+
+import javax.jms.JMSException;
+import javax.jms.ObjectMessage;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.opensmartgridplatform.adapter.protocol.dlms.application.services.BundleService;
+import org.opensmartgridplatform.adapter.protocol.dlms.application.services.DomainHelperService;
+import org.opensmartgridplatform.adapter.protocol.dlms.application.services.ThrottlingService;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDeviceBuilder;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ConnectionException;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.OsgpExceptionConverter;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DeviceResponseMessageSender;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.RetryHeaderFactory;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.BundleMessagesRequestDto;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class BundleMessageProcessorTest {
+
+    @Mock
+    private ObjectMessage message;
+
+    @Mock
+    private OsgpExceptionConverter osgpExceptionConverter;
+
+    @Mock
+    private RetryHeaderFactory retryHeaderFactory;
+
+    @Mock
+    private DeviceResponseMessageSender responseMessageSender;
+
+    @Mock
+    private DomainHelperService domainHelperService;
+
+    @Mock
+    private ThrottlingService throttlingService;
+
+    @Mock
+    private DlmsConnectionHelper dlmsConnectionHelper;
+
+    @Mock
+    private BundleService bundleService;
+
+    @Mock
+    private DlmsConnectionManager dlmsConnectionManager;
+
+    @Mock
+    private DlmsMessageListener messageListener;
+
+    @InjectMocks
+    private BundleMessageProcessor messageProcessor;
+
+    private DlmsDevice dlmsDevice = new DlmsDeviceBuilder().withHls5Active(true).build();
```

Comment:
minor: maybe put this in the setUp method so all init stuff is in the same place.

Reply:
yes